module_utils/network_lsr/argument_validator.py

```diff
@@ -311,11 +311,17 @@ def _validate_impl(self, value, name):
         raise ValidationError(name, "must be an boolean but is '%s'" % (value))
 
 
-class ArgValidatorDeprecated:
+class ArgValidatorDeprecated(ArgValidator):
     def __init__(self, name, deprecated_by):
-        self.name = name
+        ArgValidator.__init__(self, name, default_value=ArgValidator.MISSING)
         self.deprecated_by = deprecated_by
 
+    def _validate_impl(self, value, name):
+        # The Deprecated validator is not a full fledged validator that can
+        # also validate settings. All it does is a marker to rename/alias
+        # the validator with another one.
```

Comment:
This is a description of the class and therefore should be in a docstring for the class in my opinion, what do you think?

Reply:
my problem with this (as before), is that it's hard to write a good, complete docstring for the class. I also find that many docstrings are redundant and line-noise, and the human would better read the code instead. So, when I would set up to write one, it rather be good.

The requirements for a good code comment are smaller, but also they can more specifically highlight a certain, immediate point. Instead of the much larger goal to document the entire class.

So, when making suggestions how to improve the documentation, please be very specific (or provide a patch), because I lack the imagination for how a good class docstring could be.